### ISSUE TITLE
Matcher to check if a list contains an item with a cetain property set to a value

### ISSUE THREAD

**Issue by Dambakk** [kotest/kotest#1039]
Hi!

I have a feature request. I have a list of objects and would like to verify that one of the items has a specific property with a certain value. Kind of like this: 

`usersList shouldContainItemWith(User::id equalTo adminUserId)`

This could also be inverted to verify that the items do not contain an item with a certain value, e.g. to verify that something has been removed from the list of items.

Is there an existing way of doing this? If not, do you want me to create a pull request?

**Comment by sksamuel** [comment]
Try `shouldExist`

**Comment by Dambakk** [comment]
You are right,

`usersList shouldExist { user -> user.id == adminUserId }` 

achieves the same result. Although I feel like the code my first comment is more readable, it is not worth a pull request.

Thanks!

**Comment by sksamuel** [comment]
You could do `usersList.forSome { it.id shouldBe adminUserId }` as well

**Comment by Dambakk** [comment]
`forSome` would not give the expected result in our situation, it will not succeed if our admin user is the only user in the list.

On another note, I notice that description for both `shouldExist` and `forSome` are missing from the list of matchers found here: https://github.com/kotlintest/kotlintest/blob/master/doc/matchers.md

Is it coming or should it be created a new issue?

**Comment by sksamuel** [comment]
forSome is documented on the inspectors page as it's not a matcher.
It should work for your use case, even if the admin is the only user.

**Comment by Dambakk** [comment]
Ah, I see, I found it under inspectors. But as it is described there: "if NONE pass or ALL pass then we consider that a failure" - meaning that if the admin user is the only user in the list, it will fail. 

`shouldExists` was the way to go in our case 👍 But I still miss a description of it under Collections on the matchers page. I looked there when writing my test and ended up writing my own matcher which did the same. :)
